spec/requests/admin/bulk_assign_role_spec.rb

```diff
@@ -0,0 +1,118 @@
+require "rails_helper"
+
+RSpec.describe "Admin::BulkAssignRole" do
+  let(:admin) { create(:user, :super_admin) }
+  let(:user1) { create(:user) }
+  let(:user2) { create(:user) }
+  let(:user3) { create(:user) }
+  let(:usernames_string) { "#{user1.username}, #{user2.username}, #{user3.username}" }
+  let(:role) { "Trusted" }
+  let(:note_for_current_role) { "Assigned Trusted role" }
+  let(:extra_space_usernames_string) { "#{user1.username},  #{user2.username},  #{user3.username}" }
+  let(:invalid_role) { "Invalid Role" }
+
+  describe "POST /admin/member_manager/bulk_assign_role" do
+    before do
+      sign_in admin
+    end
+
+    it "adds trusted role successfully" do
+      params = { usernames: usernames_string, role: role, note_for_current_role: note_for_current_role }
+      post admin_bulk_assign_role_path, params: params
+
+      expect(response).to redirect_to(admin_bulk_assign_role_index_path)
+      expect(flash[:success]).to eq(I18n.t("admin.bulk_assign_role_controller.success_message"))
+
+      expect(user1.roles.count).to eq(1)
+      expect(user1.roles.last.name).to eq("trusted")
```

Comment:
Minor: Abstracting the role into `let(:role)` is good, but we _ought_ to be able to re-use it here in the expectation as well.